Material on managing access to the repository (#15)
empa-scientific-it/how-to-git: src/content/docs/05-collaborate.mdx

### Manage access to the repository

> The fact that repository can be private or public already influences the basic aspects of repository's access management, as discussed in the [Public vs Private repositories](../repository/#public-vs-private-repositories) section, so we will not repeat it here.

Reviewer: ```suggestion
The fact that a repository can be private or public already influences its basic access management, as discussed in the [Public vs Private repositories](../repository/#public-vs-private-repositories) section, so we will not repeat it here.
```

> However, there are more advanced features that allow you to manage access to the repository in more detail.

Reviewer: ```suggestion
However, there are advanced features that allow you to manage access to the repository in more detail.
```

> A GitHub repository can belong either to an individual user or to a GitHub organization (a group of users):

Reviewer: maybe you could explain this to support the explanation in Line 51 in the ### Public vs Private repositories

  - If the repository belongs to an individual, that user is the owner and has full administrative control.

  - If the repository belongs to an organization, then the organization owners are the admins of all repositories within the organization.

Let's focus on the repository level access first.
To invite collaborators to a privately owned repository, you can open the repository on GitHub, click on the "Settings" tab, and then select "Collaborators" from the left sidebar.
There, you can find "Manage access" section, where you can invite collaborators by their GitHub username.

> When you invite a collaborator, they will receive an email notification and will need to accept the invitation to gain access to the repository.

Reviewer: will they receive an email notification in any case, or just if they set this settings right?

Author: In that case, it is always an email

After the invitation is accepted, the new collaborator will be assigned a "collaborator role" with write access to the repository.

> You cannot assign different roles to collaborators in this case.

Reviewer: I'm not completely sure what you mean, but I guess it's the following?
```suggestion
On the repository level you can only assign write access to collaborators.
```

Author: fixed in https://github.com/empa-scientific-it/how-to-git/pull/15/commits/81ad18a3783096954ea61c9353c5bce80aa2afab

> The situation is a bit different for repositories that belong to an organization.

Reviewer: It's only like that, if the person adressed here is also belonging to the organization, right? should this be mentioned?
I understood, if you are part of an organization, you are also an admin of the organizations repositories. So are the rules described here, if you are an admin of an repository?

Author: fixed in https://github.com/empa-scientific-it/how-to-git/pull/15/commits/f2cd7466a40dccaac32b30eb7bfc2544a2cf5927

To invite people to a repository that belongs to an organization, you can click on the "Settings" tab of the repository, then select "Collaborators and teams" from the left sidebar, and then click on the "Add people" button.
You can then search for the GitHub username of the person you want to invite and select them.
After selecting the person, you can assign them a role for the repository (see below) and click "Invite".

> The role can be one of the following: `Read`, `Triage`, `Write`, `Maintain`, or `Admin`.

Reviewer: I don't understand, why there are so many options more if there is an organisation involved

Author: I don't know exactly, but I assume it is because organisation implies different levels of collaboration among the participants. Do you want me to put some text here?
